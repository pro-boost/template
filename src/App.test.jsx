// App.test.jsx

import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
  it("renders All Micght", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("renders Endavor after button click", async () => {
    const user = userEvent.setup();
    render(<App />);
    const button = screen.getByRole("button");
    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/Endavor/i);
  });
});
