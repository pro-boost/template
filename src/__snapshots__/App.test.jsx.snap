// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders All Micght 1`] = `
<div>
  <h1>
    All Might
  </h1>
  <button>
    change hero
  </button>
</div>
`;

exports[`App component > renders All Might 1`] = `
<div>
  <h1>
    All Might
  </h1>
  <button>
    Change Hero
  </button>
</div>
`;

exports[`App component > renders magnificent monkeys 1`] = `
<div>
  <h1>
    All Might
  </h1>
  <button>
    Change Hero
  </button>
</div>
`;
